[Logic] `az logic workflow`: Fix the issue with `parameters` in `--definition` not working

Jing-song opened the pull request:
---

Fix the issue with `parameters` in `--definition` not working
close https://github.com/Azure/azure-cli/issues/27982
aaz model https://github.com/Azure/aaz/pull/379

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
<!--- Please provide the related command with az {command} if you can, so that we can quickly route to the related person to review. --->
`az logic workflow`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).


### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

yonzhan commented:
<!-- add-pr-comment:add-pr-comment -->

Logic

jsntcy commented:
> ❌Azure CLI Extensions Breaking Change Test
> > ❌logic
> > 
> > 
> > 
> > rule
> > cmd_name
> > rule_message
> > suggest_message
> > 
> > 
> > 
> > 
> > ❌ [1009 - ParaPropRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1009.md)
> > logic workflow create
> > cmd `logic workflow create` update parameter `definition`: removed property `nargs=?`
> > please add back property `nargs=?` for parameter `definition` of cmd `logic workflow create`
> > 
> > 
> > ❌ [1009 - ParaPropRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1009.md)
> > logic workflow update
> > cmd `logic workflow update` update parameter `definition`: removed property `nargs=?`
> > please add back property `nargs=?` for parameter `definition` of cmd `logic workflow update`

@Jing-song, are these breaking changes expected?

Jing-song commented:
> > ❌Azure CLI Extensions Breaking Change Test
> > > ❌logic
> > > rule
> > > cmd_name
> > > rule_message
> > > suggest_message
> > > ❌ [1009 - ParaPropRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1009.md)
> > > logic workflow create
> > > cmd `logic workflow create` update parameter `definition`: removed property `nargs=?`
> > > please add back property `nargs=?` for parameter `definition` of cmd `logic workflow create`
> > > ❌ [1009 - ParaPropRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1009.md)
> > > logic workflow update
> > > cmd `logic workflow update` update parameter `definition`: removed property `nargs=?`
> > > please add back property `nargs=?` for parameter `definition` of cmd `logic workflow update`
> 
> @Jing-song, are these breaking changes expected?

Yes, these are generated by codegen
